version: '3.9'

services:
  mysql:
    image: mysql:8.0
    container_name: jobready
    environment:
      MYSQL_ROOT_PASSWORD: root
      MYSQL_DATABASE: jobready
    ports:
      - "3306:3306"
    volumes:
      - mysql_data:/var/lib/mysql
    networks:
      - app-network
    healthcheck:
      test: ["CMD", "mysqladmin", "ping", "-h", "localhost"]
      timeout: 20s
      retries: 10

  discovery-service:
    build:
      context: .
      dockerfile: discovery-service/Dockerfile
    container_name: discovery-service
    env_file:
      - ./discovery-service/.env
    ports:
      - "8761:8761"
    networks:
      - app-network

  gateway-service:
    build:
      context: .
      dockerfile: gateway-service/Dockerfile
    container_name: gateway-service
    env_file:
      - ./gateway-service/.env
    ports:
      - "8080:8080"
    depends_on:
      - discovery-service
    networks:
      - app-network

  auth-service:
    build:
      context: .
      dockerfile: auth-service/Dockerfile
    container_name: auth-service
    env_file:
      - ./auth-service/.env
    ports:
      - "8082:8082"
    depends_on:
      - discovery-service
    networks:
      - app-network

  user-service:
    build:
      context: .
      dockerfile: user-service/Dockerfile
    container_name: user-service
    env_file:
      - ./user-service/.env
    ports:
      - "8083:8083"
    depends_on:
      mysql:
        condition: service_healthy
      discovery-service:
        condition: service_started
    networks:
      - app-network

volumes:
  mysql_data:

networks:
  app-network:
    driver: bridge
